core/src/util.jl

```diff
@@ -692,3 +706,28 @@ function set_is_pid_controlled!(p::Parameters)::Nothing
     end
     return nothing
 end
+
+function has_external_demand(
+    graph::MetaGraph,
+    node_id::NodeID,
+    node_type::Symbol,
+)::Tuple{Bool, Union{NodeID, Nothing}}
+    control_inneighbors = inneighbor_labels_type(graph, node_id, EdgeType.control)
+    for id in control_inneighbors
+        if graph[id].type == node_type
+            return true, id
+        end
+    end
+    return false, nothing
+end
+
+function Base.get(
```

Comment:
Did you intend to add a method to `getindex` instead so you can do `A[i]`?

Reply:
Looks like I don't use this at all anymore

Reply:
Ah no I use it, but the use case is that I want to be able to get a constraint by node ID or get `nothing` if the constraint does not exist.